tests/parser/numeric-separators.ts

```diff
@@ -16,3 +19,15 @@
 
 0x23_23_23_; // 6188
 0x23__23_23; // 6189
+
+1000_.1234; // 6188
+1000._1234; // 6188
+1000.1234_; // 6188
+
+10__00.1234; // 6189
+1000.12__34; // 6189
+
+1_e2; // 6188
+1e_2; // 6188
+1e2_; // 6188
+1e-1__0; // 6189
```

Comment:
nice! Could you also add this invalid cases:
```ts
_1.0
0_0.0
0_0.0_0
```
and valid case:
```ts
0.0_0
```

Reply:
I am a little confused about some of these error cases.

How would you recommend testing `_1.0`? It seems like it wouldn't be parsed as a float - it would be an identifier `_1` and a float `0`. Should I expect that as output?

All of these seem to be accepted by the typescript compiler and interpreted as decimal floats:
```
0_0.0
0_0.0_0
0_0e0_0
```
Could you point me towards the spec that AssemblyScript is following for this feature? Or could you summarize the specific problem with these examples? Is it just that `0_` is an invalid prefix of any of the fragments?

Thanks!

Reply:
Hmm, really TS accept this. However by spec leading `0_` should fail: https://github.com/tc39/test262/blob/master/test/language/literals/numeric/numeric-separators/numeric-separator-literal-nzd-nsl-dds-leading-zero-err.js

Reply:
> How would you recommend testing _1.0

Yeah, it should start parsing as identifier, so I removed this from suggestions